README.md

```diff
@@ -308,13 +314,176 @@ securely.
 
 **Congratulations! You are ready to secure your own apps with Conjur.**
 
+### Using persistent Conjur configuration
+
+With small variations to the steps outlined above, it is possible to set
+up a Conjur OSS environment that retains Conjur configuration or state
+across Docker container restarts. Using the steps outlined below, a
+Conjur OSS environment can be set up that uses a local directory on
+the host to persist Conjur configuration across container restarts.
+
+#### Set up a Conjur OSS environment with persistence
+
+1. If you are already running the Conjur OSS quickstart environment without
+   persistence, bring down the associated containers:
+
+   ```
+   docker-compose down
+   ```
+
+1. Create a directory for storing persistent state. For example:
+
+   ```
+   mkdir temp-db-data
+   chmod 700 temp-db-data
```

Comment:
Does this need to be 777?

Reply:
Actually, it doesn't seem to matter what the permissions are set to for this directory. Looks like docker-compose will change the permissions for you when it gets mounted as a container volume:

```
dane@dane-vbox:~/cyberark2/conjur-quickstart$ mkdir temp-db-data
dane@dane-vbox:~/cyberark2/conjur-quickstart$ ls -ld temp-db-data
drwxr-xr-x 2 dane dane 4096 Aug 14 09:45 temp-db-data
dane@dane-vbox:~/cyberark2/conjur-quickstart$ docker-compose up -d
Creating network "conjur-quickstart_default" with the default driver
Creating postgres_database ... done
Creating bot_app           ... done
Creating openssl           ... done
Creating conjur_server     ... done
Creating nginx_proxy       ... done
Creating conjur_client     ... done
dane@dane-vbox:~/cyberark2/conjur-quickstart$ ls -ld temp-db-data/
drwx------ 18 vboxadd dane 4096 Aug 14 09:48 temp-db-data/
dane@dane-vbox:~/cyberark2/conjur-quickstart$ 
```

So maybe I should delete the line `chmod 700 temp-db-data` and just add a note saying that the permissions on the file will be changed by docker-compose when directory gets mounted in the container.

Reply:
I see the same behavior on Mac.

Reply:
I think you may need to try writing data to the DB then fetching for this to show problems if there's any.

Reply:
It appears to work fine. Here's a session that includes restart:
```
dane@dane-vbox:~/cyberark2/conjur-quickstart$ mkdir temp-db-data
dane@dane-vbox:~/cyberark2/conjur-quickstart$ ls -ld temp-db-data/
drwxr-xr-x 2 dane dane 4096 Aug 14 10:28 temp-db-data/
dane@dane-vbox:~/cyberark2/conjur-quickstart$ docker-compose up -d
Creating network "conjur-quickstart_default" with the default driver
Creating bot_app           ... done
Creating openssl           ... done
Creating postgres_database ... done
Creating conjur_server     ... done
Creating nginx_proxy       ... done
Creating conjur_client     ... done
dane@dane-vbox:~/cyberark2/conjur-quickstart$ ls -ld temp-db-data
drwx------ 18 vboxadd dane 4096 Aug 14 10:29 temp-db-data
dane@dane-vbox:~/cyberark2/conjur-quickstart$ docker-compose exec conjur conjurctl account create myConjurAccount > admin_data
dane@dane-vbox:~/cyberark2/conjur-quickstart$ docker-compose exec client conjur init -u conjur -a myConjurAccount
Wrote configuration to /root/.conjurrc
dane@dane-vbox:~/cyberark2/conjur-quickstart$ docker-compose exec client conjur authn login -u admin
Please enter admin's password (it will not be echoed): 
Logged in
dane@dane-vbox:~/cyberark2/conjur-quickstart$ docker-compose exec client conjur policy load root policy/BotApp.yml > my_app_data
dane@dane-vbox:~/cyberark2/conjur-quickstart$ docker-compose exec client conjur list variables
[
  "myConjurAccount:policy:root",
  "myConjurAccount:policy:BotApp",
  "myConjurAccount:user:Dave@BotApp",
  "myConjurAccount:host:BotApp/myDemoApp",
  "myConjurAccount:variable:BotApp/secretVar"
]
dane@dane-vbox:~/cyberark2/conjur-quickstart$ docker-compose down
Stopping conjur_client     ... done
Stopping nginx_proxy       ... done
Stopping conjur_server     ... done
Stopping postgres_database ... done
Stopping bot_app           ... done
Removing conjur_client     ... done
Removing nginx_proxy       ... done
Removing conjur_server     ... done
Removing postgres_database ... done
Removing bot_app           ... done
Removing openssl           ... done
Removing network conjur-quickstart_default
dane@dane-vbox:~/cyberark2/conjur-quickstart$ docker-compose up -d
Creating network "conjur-quickstart_default" with the default driver
Creating bot_app           ... done
Creating postgres_database ... done
Creating openssl           ... done
Creating conjur_server     ... done
Creating nginx_proxy       ... done
Creating conjur_client     ... done
dane@dane-vbox:~/cyberark2/conjur-quickstart$ docker-compose exec client conjur init -u conjur -a myConjurAccount
Wrote configuration to /root/.conjurrc
dane@dane-vbox:~/cyberark2/conjur-quickstart$ docker-compose exec client conjur authn login -u admin
Please enter admin's password (it will not be echoed): 
Logged in
dane@dane-vbox:~/cyberark2/conjur-quickstart$ docker-compose exec client conjur list variables
[
  "myConjurAccount:policy:root",
  "myConjurAccount:policy:BotApp",
  "myConjurAccount:user:Dave@BotApp",
  "myConjurAccount:host:BotApp/myDemoApp",
  "myConjurAccount:variable:BotApp/secretVar"
]
dane@dane-vbox:~/cyberark2/conjur-quickstart$ 
```

Reply:
I verified on Mac also: File permissions get changed to 700 when the directory is host mounted, and writing/persisting database appears to work fine.